freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/user/model/SynchronizeOperationRequestBase.java

```diff
@@ -3,12 +3,16 @@
 import java.util.HashSet;
 import java.util.Set;
 
+import com.sequenceiq.authorization.annotation.ResourceObjectField;
+import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
+import com.sequenceiq.authorization.resource.AuthorizationVariableType;
 import com.sequenceiq.freeipa.api.v1.freeipa.user.doc.UserModelDescriptions;
 
 import io.swagger.annotations.ApiModelProperty;
 
 public class SynchronizeOperationRequestBase {
     @ApiModelProperty(value = UserModelDescriptions.USERSYNC_ENVIRONMENT_CRNS)
+    @ResourceObjectField(action = AuthorizationResourceAction.EDIT_ENVIRONMENT, variableType = AuthorizationVariableType.CRN_LIST)
```

Comment:
the field name `environments` is misleading in this case, i assumed that with that field I can tell which environments I want to apply the sync for.

Reply:
yes. this is one of many reasons why I would like to re-do the user sync APIs once I get a chance

Reply:
sorry i removed my comment because it was only partly true, but yes, it would be nice if at a time we/you would be able to redesign this to make it clearer